apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -0,0 +1,661 @@
+
+      ]},
+     {internal_endpoints, [sequence],
+      [
+       broken_spend_tx,
+       miner_pub_key
+      ]}
+    ].
+
+suite() ->
+    [].
+
+init_per_suite(Config) ->
+    ok = application:ensure_started(erlexec),
+    DataDir = ?config(data_dir, Config),
+    TopDir = aecore_suite_utils:top_dir(DataDir),
+    Config1 = [{symlink_name, "latest.http_endpoints"},
+               {top_dir, TopDir},
+               {test_module, ?MODULE}] ++ Config,
+    aecore_suite_utils:make_shortcut(Config1),
+    ct:log("Environment = ~p", [[{args, init:get_arguments()},
+                                 {node, node()},
+                                 {cookie, erlang:get_cookie()}]]),
+    aecore_suite_utils:create_configs(Config1, #{<<"chain">> =>
+                                                 #{<<"persist">> => false}}),
+    aecore_suite_utils:make_multi(Config1, [?NODE]),
+    Config1.
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_group(_, Config) ->
+    [{nodes, [aecore_suite_utils:node_tuple(?NODE)]} | Config].
+    
+
+end_per_group(_Group, _Config) ->
+    ok.
+
+init_per_testcase(_Case, Config) ->
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(aecore_suite_utils:node_name(?NODE)),
+    ct:log("testcase pid: ~p", [self()]),
+    [{tc_start, os:timestamp()}|Config].
+
+end_per_testcase(_Case, Config) ->
+    aecore_suite_utils:stop_node(?NODE, Config),
+    Ts0 = ?config(tc_start, Config),
+    ct:log("Events during TC: ~p", [[{N, aecore_suite_utils:all_events_since(N, Ts0)}
+                                     || {_,N} <- ?config(nodes, Config)]]),
+    ok.
+
+%% ============================================================
+%% Test cases
+%% ============================================================
+correct_ping(_Config) ->
+    PingObj = rpc(aec_sync, local_ping_object, []),
+    {ok, 200, _} = post_ping(maps:put(<<"source">>, unique_peer(), PingObj)),
+    ok.
+
+broken_pings(_Config) ->
+    % no 'source'
+    {ok, 400, _} = post_ping(#{}),
+    % no ping obj data
+    {ok, 400, _} = post_ping(#{<<"source">> => unique_peer()}),
+    PingObj = rpc(aec_sync, local_ping_object, []),
+    % wrong genesis hash 
+    WrongGenHashPing = maps:merge(PingObj, #{<<"source">> => unique_peer(),
+                                             <<"genesis_hash">> => <<"foo">>}),
+    {ok, 404, #{<<"reason">> := <<"Different genesis blocks">>}} =
+        post_ping(WrongGenHashPing),
+    ok.
+
+blocked_ping(_Config) ->
+    Peer = unique_peer(),
+    PingObj = rpc(aec_sync, local_ping_object, []),
+    WrongGenHashPing = maps:merge(PingObj, #{<<"source">> => Peer,
+                                             <<"genesis_hash">> => <<"foo">>}),
+    {ok, 404, _} = post_ping(WrongGenHashPing),
+    % node is blocked now 
+    {ok, 404, #{<<"reason">> := <<"Not allowed">>}} =
+        post_ping(maps:put(<<"source">>, Peer, PingObj)),
+    ok.
+
+get_top_empty_chain(_Config) ->
```

Comment:
It feels renaming would help, because this test appears to assume only that the top of the chain returned by the HTTP layer is the same as the one returned by Erlang. So:
* It assumes no concurrent modification of the chain (is automining stopped? does not have peers that could push block?) that may be symptom of fragility of test;
* It does not check that chain is "empty" i.e. with only genesis (or if it does, that is not clear to me).